src/layers/core/choropleth-layer/choropleth-layer.js

```diff
@@ -210,3 +207,89 @@ export default class ChoroplethLayer extends BaseLayer {
   }
 
 }
+
+// convert geojson to choropleths
+// @param {object} data - geojson object
+// @returns {array.array.array.point} array of geojson polygon-like coordinates
+function extractChoropleths(data) {
+  const normalizedGeojson = normalize(data);
+  const result = [];
+
+  normalizedGeojson.features.map((feature, featureIndex) => {
+    const choropleths = featureToChoropleths(feature);
+    choropleths.forEach(choropleth => {
+      choropleth.featureIndex = featureIndex;
+    });
+    result.push(...choropleths);
+  });
+  return result;
+}
+
+// normalize the coordinates array of a geojson feature
+// @param {object} feature - geojson feature object
+// @returns {array.array.array.point} array of geojson polygon-like coordinates
+function featureToChoropleths(feature) {
+  const {coordinates, type} = feature.geometry;
+  let choropleths;
+
+  switch(type) {
+  case 'MultiPolygon':
+    choropleths = coordinates;
+    break;
+  case 'Polygon':
+    choropleths = [coordinates];
+    break;
+  case 'LineString':
```

Comment:
OK for now, but long-term, should line strings be handled here? They are not closed paths and should not be filled, so I assume this will not always give expected results? 

- It's almost like we should have one PolygonLayer, one LineStringLayer and one PointLayer.
- Even better, we should change deck.gl so that one layer can have multiple primitives. That would be a great addition for deck.gl v3.